framework/helpers/json.go

```diff
@@ -0,0 +1,45 @@
+package helpers
+
+import (
+	"encoding/json"
+	"sort"
+	"strings"
+
+	"gopkg.in/launchdarkly/go-sdk-common.v2/ldvalue"
+)
+
+// AsJSON is just a shortcut for calling json.Marshal and taking only the first result.
+func AsJSON(value interface{}) []byte {
+	ret, _ := json.Marshal(value)
```

Comment:
Yes.

Reply:
That is... in this case it is safe, and in general it is almost always safe to return nil for a `[]byte` result.

The reason is that there are only a limited number of things you can do with a byte slice:

1. You could get its length with `len`.
2. You could convert it to a string.
3. You could try to get an individual byte element from it.
4. You could get a sub-slice of it

1 and 2 are literally _always_ safe; converting `nil` to a string just gives you `""`, and using `len` gives you zero.

3 and 4 are potentially unsafe with nil— but only in the same sense that they're potentially unsafe with a non-nil slice too, because they can panic if you use an out-of-range index. So you would always want to do a length check first, in which case again `nil` behaves just the same as a zero-length slice.

Reply:
So, even though I said "almost always", I'm actually having trouble thinking of any case where it is not safe. I guess maybe if you're doing direct pointer manipulation stuff with the `unsafe` package, there might be some situation where treating a nil slice value as a pointer could get you in trouble. So there might be an unsafe situation if you're using the package that is actually named that.